pkg/virtual/apiexport/builder/build.go

```diff
@@ -0,0 +1,174 @@
+package builder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/kcp-dev/logicalcluster"
+
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/virtual/apiexport/controllers/apireconciler"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework"
+	virtualdynamic "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apidefinition"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apiserver"
+	dynamiccontext "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/context"
+)
+
+const VirtualWorkspaceName string = "apiexport"
+
+func BuildVirtualWorkspace(
+	rootPathPrefix string,
+	dynamicClusterClient dynamic.ClusterInterface,
+	kcpClusterClient kcpclient.ClusterInterface,
+	wildcardKcpInformers kcpinformer.SharedInformerFactory,
+) framework.VirtualWorkspace {
+	virtualWorkspacePathPrefix := path.Join(rootPathPrefix, VirtualWorkspaceName)
+
+	if !strings.HasSuffix(virtualWorkspacePathPrefix, "/") {
+		virtualWorkspacePathPrefix += "/"
+	}
+
+	readyCh := make(chan struct{})
+
+	return &virtualdynamic.DynamicVirtualWorkspace{
+		Name: VirtualWorkspaceName,
+
+		RootPathResolver: func(path string, requestContext context.Context) (accepted bool, prefixToStrip string, completedContext context.Context) {
+			completedContext = requestContext
+
+			if !strings.HasPrefix(path, virtualWorkspacePathPrefix) {
+				return
+			}
+
+			// /services/apiexport/$cluster/$apiExport/$identity/clusters/...
+
+			withoutRootPathPrefix := strings.TrimPrefix(path, virtualWorkspacePathPrefix)
+
+			parts := strings.SplitN(withoutRootPathPrefix, "/", 3)
+			if len(parts) < 3 {
+				return
+			}
+
+			apiExportClusterName, apiExportName := parts[0], parts[1]
+			if apiExportClusterName == "" {
+				return
+			}
+			if apiExportName == "" {
+				return
+			}
+
+			realPath := "/"
+			if len(parts) > 2 {
+				realPath += parts[2]
+			}
+
+			cluster := genericapirequest.Cluster{Name: logicalcluster.Wildcard, Wildcard: true}
```

Comment:
Is this something we want to do here? I'm still not sure we should be messing around with what default semantics are in VW in general, just for our own sanity's sake ... but also in this particular case, why?

Reply:
that's code copied from Andy's PR. Have to look at that part. I agree that seeing WithCluster here is irritating.

Reply:
Can remove. Was probably part of my thought process.

Reply:
Moved here: https://github.com/kcp-dev/kcp/pull/1144